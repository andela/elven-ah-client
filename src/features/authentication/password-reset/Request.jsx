import React from 'react';
import { PropTypes } from 'prop-types';

/* eslint-disable react/no-array-index-key */
const Request = ({
  handleChange, values, handleRequest, errors,
}) => (
  <div className="mx-auto centered col-xl-4 col-lg-4 col-md-6 col-sm-6 col-xs-6">
    <form className="form-login" onSubmit={handleRequest}>
      <h4 className="text-centre">Enter your registered email to request a password reset link</h4>
      <div className="error">{errors.message}</div>
      <input
        onChange={handleChange}
        type="email"
        className="form-control form-input"
        id="email"
        value={values.email}
        placeholder="Enter your email address"
        required
      />
      {errors.email ? errors.email.map((error, index) => <span className="error" key={index}>{error}</span>) : ''}
      <button className="btn mx-auto btn-block auth-submit-btn" type="submit">Request Password Reset</button>
    </form>
  </div>
);

Request.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleRequest: PropTypes.func.isRequired,
  errors: PropTypes.shape({}).isRequired,
  values: PropTypes.shape({}).isRequired,
};

export default Request;
