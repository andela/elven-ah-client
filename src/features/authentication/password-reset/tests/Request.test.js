import React from 'react';
import { shallow } from 'enzyme';

import Request from '../Request';

const mockFunction = jest.fn();
const errors = {
  email: ['The password field is required'],
};
const values = {
  email: '',
  confirmPassword: '',
  password: '',
};
describe('Test the Request password reset presentation component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Request
      handleChange={mockFunction}
      handleRequest={mockFunction}
      resetLinkError="resetLinkError"
      errors={errors}
      values={values}
    />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without errors', () => {
    const wrapper = shallow(<Request
      handleChange={mockFunction}
      handleRequest={mockFunction}
      errors={errors}
      resetLinkError="resetLinkError"
      values={values}
    />);
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(1);
  });
});
