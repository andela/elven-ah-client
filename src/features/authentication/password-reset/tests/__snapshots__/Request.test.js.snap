// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Request password reset presentation component renders without crashing 1`] = `
<div
  className="mx-auto centered text-center col-xl-4 col-lg-4 col-md-6 col-sm-6 col-xs-6"
>
  <form
    className="form-login"
    onSubmit={[Function]}
  >
    <h4>
      Enter your registered email to request a password reset link
    </h4>
    <div
      className="form-label-group mb-3"
    >
      <input
        className="form-control form-input"
        id="email"
        onChange={[Function]}
        placeholder="Enter your registered email"
        required={true}
        type="email"
        value=""
      />
      <span
        className="invalid-feedback"
      />
      <span
        className="invalid-feedback"
      >
        resetLinkError
      </span>
      <span
        className="invalid-feedback"
        key="0"
      >
        The password field is required
      </span>
    </div>
    <span
      className="error"
      key="0"
    >
      The password field is required
    </span>
    <button
      className="btn mx-auto btn-block auth-submit-btn"
      type="submit"
    >
      Request Password Reset
    </button>
  </form>
</div>
`;
